### PR TITLE
hw-2.2

### PR DESCRIPTION
**Pull request by Djeckkenni**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -1,23 +1,51 @@
...
 /**
  * Реалізуйте методи для знаходження головної та побічної діагоналі квадратної матриці,
  * максимального та мінімального елементів.
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] diagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix[i].length; j++) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: try to solve with only one loop

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -1,23 +1,51 @@
...
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        int j = matrix.length - 1;
+        for (int i = 0; i < matrix.length; i++) {
+            counterDiagonal[i] += matrix[i][j];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: why do you use `+=`?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -1,23 +1,51 @@
...
+            j--;
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int[] min = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            Arrays.sort(matrix[i]);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: can you solve without Arrays class everywhere?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -1,23 +1,51 @@
...
+            min[i] += matrix[i][0];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: why `+=`?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 5]
```diff
@@ -1,23 +1,51 @@
...
+        Arrays.sort(min);
+        return min[0];
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int[] max = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            Arrays.sort(matrix[i]);
+            max[i] += matrix[i][matrix.length - 1];
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: same

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 3 above)
Comment (Djeckkenni) [reply src/main/java/core/basesyntax/MatrixOperation.java]: добавила свой метод сортировки

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (Djeckkenni) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (Djeckkenni) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Убрала +=

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 4 above)
Comment (Djeckkenni) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Done

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 5 above)
Comment (Djeckkenni) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Done

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 6]
```diff
@@ -4,20 +4,55 @@
...
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        int j = matrix.length - 1;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: can you solve without this variable?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 7]
```diff
@@ -4,20 +4,55 @@
...
+        int[] max = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            sortMatrix(matrix[i]);
+            max[i] = matrix[i][matrix.length - 1];
+        }
+        sortMatrix(max);
+        return max[max.length - 1];
+    }
+
+    public int[] sortMatrix(int[] matrix) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/MatrixOperation.java]: please, make a solution without sorting the matrix. sorting has a lot of iterations which is not optimal

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**